youtube_dl/extractor/weiqitv.py

```diff
@@ -0,0 +1,89 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import datetime
+import base64
+
+from .common import InfoExtractor
+from ..utils import (
+    parse_iso8601,
+    sanitized_Request,
+)
+
+
+class WeiqitvIE(InfoExtractor):
+    IE_DESC = 'WQTV'
+    _VALID_URL = r'http://www\.weiqitv\.com/index/video_play\?videoId=(?P<id>[A-Za-z0-9]+)'
+
+    _TESTS = [{
+        'url': 'http://www.weiqitv.com/index/video_play?videoId=53c744f09874f0e76a8b46f3',
+        'md5': '26450599afd64c513bc77030ad15db44',
+        'info_dict': {
+            'id': '53c744f09874f0e76a8b46f3',
+            'ext': 'mp4',
+            'title': '2013年度盘点',
+            'timestamp': 1445922586,
+            'upload_date': '20151027',
+        },
+        'params': {
+            'format': '[width=1920]',
+        },
+    }, {
+        'url': 'http://www.weiqitv.com/index/video_play?videoId=567379a2d4c36cca518b4569',
+        'info_dict': {
+            'id': '567379a2d4c36cca518b4569',
+            'ext': 'mp4',
+            'title': '民国围棋史',
+            'timestamp': 1450325068,
+            'upload_date': '20151217',
+        },
+    }, {
+        'url': 'http://www.weiqitv.com/index/video_play?videoId=5430220a9874f088658b4567',
+        'info_dict': {
+            'id': '5430220a9874f088658b4567',
+            'ext': 'mp4',
+            'title': '二路托过的手段和运用',
+            'timestamp': 1445946428,
+            'upload_date': '20151027',
+        },
+    }]
+
+    def _real_extract(self, url):
+        media_id = self._match_id(url)
+        page = self._download_webpage(url, media_id)
+
+        info_json_str = self._html_search_regex(
+            'var video = (.*});',
+            page, 'info_json_str', default=None)
+        info_json = self._parse_json(info_json_str, media_id)
+
+        play_json_param = self._html_search_regex(
+            r'letvurl.*?bcloud.html\?([^"]+)',
+            page, 'play_json_param', default=None)
+        play_json_req = sanitized_Request(
+            'http://api.letvcloud.com/gpc.php?cf=html5&sign=signxxxxx&ver=2.2&format=json&' +
```

Comment:
do you mean after find out letvurl then delegate to LetvCloudIE for the rest of work?
the LetvCloudIE should be in letv.py.